### PR TITLE
feat: notification support cssVar

### PR DESCRIPTION
**Pull request by kiner-tang**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [X] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

- #45618

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | notification support css variable theme |
| 🇨🇳 Chinese | 通知支持 css 变量主题 |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [X] Doc is updated/provided or not needed
- [X] Demo is updated/provided or not needed
- [X] TypeScript definition is updated/provided or not needed
- [X] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 70d5a05</samp>

This pull request enhances the notification component to support theme customization and dark mode using CSS variables. It introduces a new file `cssVar.ts` to generate the CSS variables, and updates the component and style modules to use the `@ant-design/cssinjs` library and the `useCSSVar` hook.

### <samp>🤖 Generated by Copilot at 70d5a05</samp>

*  Import React and hooks for generating and applying CSS variables for the notification component in `PurePanel.tsx` ([link](https://github.com/ant-design/ant-design/pull/45799/files?diff=unified&w=0#diff-8b66e26e1dc3ab97c7faf1e4f2d015c5b42d580fce0dd4c3043b7253bcdb83ccR1), [link](https://github.com/ant-design/ant-design/pull/45799/files?diff=unified&w=0#diff-8b66e26e1dc3ab97c7faf1e4f2d015c5b42d580fce0dd4c3043b7253bcdb83ccL10-R16))
*  Add a className prop and wrap the PurePanel component with the wrapCSSVar function to inject the CSS variables into the root element in `PurePanel.tsx` ([link](https://github.com/ant-design/ant-design/pull/45799/files?diff=unified&w=0#diff-8b66e26e1dc3ab97c7faf1e4f2d015c5b42d580fce0dd4c3043b7253bcdb83ccR95), [link](https://github.com/ant-design/ant-design/pull/45799/files?diff=unified&w=0#diff-8b66e26e1dc3ab97c7faf1e4f2d015c5b42d580fce0dd4c3043b7253bcdb83ccL100-R108), [link](https://github.com/ant-design/ant-design/pull/45799/files?diff=unified&w=0#diff-8b66e26e1dc3ab97c7faf1e4f2d015c5b42d580fce0dd4c3043b7253bcdb83ccL122-R132))
*  Pass the notificationClassName and rootCls class names to the Notice sub-component in `PurePanel.tsx` ([link](https://github.com/ant-design/ant-design/pull/45799/files?diff=unified&w=0#diff-8b66e26e1dc3ab97c7faf1e4f2d015c5b42d580fce0dd4c3043b7253bcdb83ccR116-R119))
*  Add a new file `cssVar.ts` that exports a function to generate a CSS variable register for the notification component ([link](https://github.com/ant-design/ant-design/pull/45799/files?diff=unified&w=0#diff-8cd30c22b25f1862f2da237884d7366f936e042343c996accef069f45c58ee8fR1-R4))
*  Import the unit function and the GenStyleFn type from the theme modules in `index.ts` ([link](https://github.com/ant-design/ant-design/pull/45799/files?diff=unified&w=0#diff-1b72cd8f976d21d83517b2c963a0b9a9bb6670e5d004ed4a7b0dd2edad3fc965L2-R2), [link](https://github.com/ant-design/ant-design/pull/45799/files?diff=unified&w=0#diff-1b72cd8f976d21d83517b2c963a0b9a9bb6670e5d004ed4a7b0dd2edad3fc965R8))
*  Modify the NotificationToken interface to accept either a number or a string for the icon and close button size properties in `index.ts` ([link](https://github.com/ant-design/ant-design/pull/45799/files?diff=unified&w=0#diff-1b72cd8f976d21d83517b2c963a0b9a9bb6670e5d004ed4a7b0dd2edad3fc965L29-R31))
*  Use the unit function and the token.calc method to perform arithmetic operations and convert numeric values to CSS units in the genNoticeStyle and prepareNotificationToken functions in `index.ts` ([link](https://github.com/ant-design/ant-design/pull/45799/files?diff=unified&w=0#diff-1b72cd8f976d21d83517b2c963a0b9a9bb6670e5d004ed4a7b0dd2edad3fc965L72-R73), [link](https://github.com/ant-design/ant-design/pull/45799/files?diff=unified&w=0#diff-1b72cd8f976d21d83517b2c963a0b9a9bb6670e5d004ed4a7b0dd2edad3fc965L101-R107), [link](https://github.com/ant-design/ant-design/pull/45799/files?diff=unified&w=0#diff-1b72cd8f976d21d83517b2c963a0b9a9bb6670e5d004ed4a7b0dd2edad3fc965L269-R275))
*  Add a type annotation to the prepareNotificationToken function using the GenStyleFn generic type in `index.ts` ([link](https://github.com/ant-design/ant-design/pull/45799/files?diff=unified&w=0#diff-1b72cd8f976d21d83517b2c963a0b9a9bb6670e5d004ed4a7b0dd2edad3fc965L262-R265))
*  Use the unit function and the token.calc method to perform arithmetic operations and convert numeric values to CSS units in the PurePanelStyle component in `pure-panel.ts` ([link](https://github.com/ant-design/ant-design/pull/45799/files?diff=unified&w=0#diff-cda3e92a534cf8b3f7539418f8e3d0e1ea4eeb84b1fbd303c43af9084685909dL14-R17))
*  Import the useCSSVar hook and wrap the Wrapper component with the wrapCSSVar function to inject the CSS variables into the root element in `useNotification.tsx` ([link](https://github.com/ant-design/ant-design/pull/45799/files?diff=unified&w=0#diff-fa8b6f08e906ed18759c056ccb358fb0b780b4ffbb0f00ed368b45c3b8efe5dfR20), [link](https://github.com/ant-design/ant-design/pull/45799/files?diff=unified&w=0#diff-fa8b6f08e906ed18759c056ccb358fb0b780b4ffbb0f00ed368b45c3b8efe5dfL39-R44))

### CODE REVIEW

File: components/notification/PurePanel.tsx [hunk 1]
```diff
@@ -97,16 +101,22 @@ const PurePanel: React.FC<PurePanelProps> = (props) => {
...
       <PurePanelStyle prefixCls={prefixCls} />
       <Notice
         {...restProps}
         prefixCls={prefixCls}
         eventKey="pure"
         duration={null}
         closable={closable}
+        className={classNames({
+          notificationClassName,
+          rootCls,
```
Comment (MadCcc) [inline components/notification/PurePanel.tsx:118]: rootCls 加在最外层就够了，这里不用加

**Comment by MadCcc** [comment]
image
有一个 nan

**Comment by kiner-tang** [comment]
> image 有一个 nan

这个看了一下，已经没有 NaN 了

**Comment by MadCcc** [comment]
`/style/placement.ts` 里还有

**Comment by kiner-tang** [comment]
> `/style/placement.ts` 里还有

image
我这边本地看没有 NaN 了呀

**Comment by MadCcc** [comment]
image
还有这里

**Comment by kiner-tang** [comment]
> image 还有这里

奇怪，我本地删除依赖重装运行 `yarn lint:style` 还是没有检测出来

**Review by MadCcc — APPROVED** [review]

**Pull request merged**